### ISSUE TITLE
Ignored users should be stopped from responding to conversation threads

### ISSUE THREAD

**Issue by Waithamai** [diaspora/diaspora#7612]
If I ignore (block) a user, they can still write responses to older, already existing conversation threads (private messages). Ignoring should stop them from responding to any such threads, just like it stops them from commenting posts. 
(Both users are on the same pod.)

Adding a screenshot of my test conversations to make my steps clearer and for your amusement.

![screenshot 2017-09-12 15:50](https://user-images.githubusercontent.com/3577985/30329592-fd276720-97d2-11e7-8f0c-c7b97e59465f.png)

**Comment by goobertron** [comment]
This seems fine for conversations where there are only two participants, where one of the two participants has ignored the other one.

What about when there are more than two participants in a conversation? Should a user be blocked from participating from the conversation when any one of the other participants ignores them? Or should their future participations in the conversation be hidden from the view of the person who has ignored them?

I don't see either solution being practical: the first is unfair, and the second is going to lead to weird communication because one of the participants is not seeing all of the comments in the conversation. And I can't think of another, better solution.

If the person who has been ignored is being abusive or plain annoying, the best solution seems to me to be to start a new conversation with the other participants, explaining to them why you're doing it, and to mute the existing one.

However, if someone can think of a workable solution to the problem in the OP, please state it!

**Comment by denschub** [comment]
Uh, those are fine points. What should we do with this? Close as wontfix? Mark as "needs discourse"? Defer our decision and wait for magic to happen?

**Comment by SuperTux88** [comment]
There is not much we can do here, because even that the conversation has an author (the person who started it), messages aren't relayables anymore since 0.6.0.0, so the author of the conversation can't reject them. But when somebody ignores another person, they shouldn't receive notifications from ignored people, I don't know if we already do that, if not we should add that.

We could also mute all conversations when somebody starts ignoring a person, but that can't be undone (for example when it's only a short ignore), and I don't think ignore should be destructive.

I think for now the "solution" from @goobertron is enough (mute the conversation manually if needed), and stop sending notifications from ignored people (if we don't do this already?).

**Comment by Waithamai** [comment]
i didn't think about conversations with more than one user. thx @goobertron! 
yes - muting might be enough then

> But when somebody ignores another person, they shouldn't receive notifications from ignored people, I don't know if we already do that, if not we should add that.
> ..and stop sending notifications from ignored people (if we don't do this already?).

notifications for conversations are still sent after ignoring someone (other notifications aren't).

**Comment by goobertron** [comment]
> What should we do with this? Close as wontfix? Mark as "needs discourse"? Defer our decision and wait for magic to happen?

I vote for turning around three times while pinching our noses and wishing very hard.

Is there a case for automatically muting conversations which have only the person ignored and the ignorer as participants? Perhaps even not that, because it is easy for the ignorer to mute is manually, and it could be a way of the ignored contacting to resolve the situation if there had been a misunderstanding.

**Comment by goobertron** [comment]
One side-issue that came up in the D* post which brought up this issue is the idea of flashing the following alert when someone mutes a conversation:

> Do you want to let other participants know that you are leaving this conversation?

What do you all think? Is this someone easy to do in the code? Is it even a useful idea?

**Comment by SuperTux88** [comment]
> What do you all think? Is this someone easy to do in the code? Is it even a useful idea?

Yes, that should be done, and I think it should be done always (automatically). The current state of conversations is really bad and inconsistent, so I think it needs some work. I don't know if it's easy, but it needs to be done, I just didn't have the time yet to give some love to conversations/messages. But that's also something that should be easier now after the federation rewrite.

You currently can't delete conversations. When everybody mutes the conversation, the last person deletes it (which is good, when everybody agrees that the conversation should be deleted, it should be deleted), but it works only on the same pod, when there are people on different pod, you can't delete it. That could be fixed by federating the mute/delete state, that's why it should be done.

I don't think we should actively notify people about that somebody left the conversation, but we should at least display it, so people know that the other person won't read it, otherwise it is bad UX in my eyes.

But I think all of this is out of scope if this issue, because this issue is about ignore. So the only thing that should be fixed here is that you don't receive notifications from a person you ignore.

**Comment by denschub** [comment]
Adjusted the title accordingly.
